alf/algorithms/predictive_representation_learner.py

```diff
@@ -94,14 +94,22 @@ def get_target_fields(self):
         return self._target_field
 
     def train_step(self, repr, state=()):
-        predicted_reward = self._decoder_net(repr)[0]
+        predicted_target = self._decoder_net(repr)[0]
+        # when the predicted target is a scalar, remove the
+        # singleton dimension to match the shape of target
+        if predicted_target.ndim == 2 and predicted_target.shape[1] == 1:
+            predicted_target = predicted_target.squeeze(1)
         return AlgStep(
-            output=predicted_reward, state=state, info=predicted_reward)
+            output=predicted_target, state=state, info=predicted_target)
```

Comment:
Sorry, I just realized that the actual shape of the target is not known here. So this still has problem if the rank of the target is greater than 1 or if the target is 1 dimensional vector. One dimensional vector may be quite common.  Perhaps we should just make sure the shape of decoder_net output is correct?

Reply:
As the ```preprocess_experience``` has the convention that the ``target`` tensor won't have a singular last dimension, so here probably we can simply use ```predicted_target = predicted_target.squeeze(1)```?

Reply:
preprocess_experience does not change the shape of the target. target can be scalar, 1d vector, regular vector, matrix, etc.

Reply:
Now use an optional ```output_tensor_spec``` for reshaping the network output.